### ISSUE TITLE
[clang] false positive when doing simultanious pack expansion of capture and arguments in variadic lambda: `[-Wunused-lambda-capture]`

### ISSUE THREAD

**Issue by r-braun** [llvm/llvm-project#116016]
Consider the following example ([Demo](https://godbolt.org/z/vx61aWTWE)):
```c++
constexpr bool test(auto... vals) {
    return [... captured = vals](auto... args){
        return ((captured == args) && ...);
    }(vals...);
}

int main() {
    static_assert(test(0,1,2));
}
```
Clang trunk (`20.0.0` as of reporting this issue) emits the warning `[-Wunused-lambda-capture]`, even though `captured` is used in the simultanious pack expansion:
```
<source>:2:17: warning: lambda capture 'captured' is not used [-Wunused-lambda-capture]
    2 |     return [... captured = vals](auto... args){
      |                 ^
```

This doesn't happen, if only `captured` is expanded ([Demo](https://godbolt.org/z/ecYqYrq5a)), or `captured` is simultaniously expanded with `vals` instead of `args` ([Demo](https://godbolt.org/z/Y5GM4za6b)).
